### ISSUE TITLE
Add colab or jupyter notebook example

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#47]
Example training scripts are already available in the examples folder, but some people might prefer to interact with a notebook.

**Comment by wconnell** [comment]
I'll put this on my list!

**Comment by KevinMusgrave** [comment]
@wconnell Let me know if you have any thoughts about the best way to visualize embeddings. I want to get rid of sklearn.tsne because it's very slow. We could switch to this instead: https://github.com/lmcinnes/umap. It seems to come with nice plotting functionality if installed with the "plot" option.

**Comment by wconnell** [comment]
UMAP is great and quickly eclipsing tSNE in some fields, I support this if speedups are significant (as they should be). 

Additionally, I'd like to point you to single-cell RNA sequencing dimensionality reduction and visualization tool, [Ivis](https://github.com/beringresearch/ivis), which uses a standard siamese nn architecture to compute embeddings. I just want to make you aware of this huge use case for metric learning in biology. Dim reduction/visualization is more of a minor goal for this library so far, but with extensibility could be a very valuable addition. Just a thought.

**Comment by KevinMusgrave** [comment]
@wconnell In v0.9.84.dev2, you can pass in a ```visualizer``` and ```visualizer_hook```. 

You can see what I mean in [this notebook I pushed to the examples folder](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/examples/VisualizationTest.ipynb). If you want to run it, you have to upgrade record-keeper. (I had to make a matplotlib import conditional because record-keeper needs to use the Agg backend, whereas jupyter needs an interactive backend.)

```pip install record-keeper==0.9.24```

The notebook is basically the same as example_MetricLossOnly, except that the train and validation sets are now class-disjoint, and some other minor changes. You can see how I use UMAP in the final cell of the notebook. Basically the user can pass in any function that has a ```fit``` and ```transform``` function. Let me know what you think of the design.

Re: ivis, would that be like an alternative to TSNE and UMAP?

**Comment by KevinMusgrave** [comment]
v0.9.84 has the changes described above. (I just realized the TSNE function provided by sklearn only has a ```fit_transform``` function so I'll allow that in the next version).

I also created a [google colab notebook](https://colab.research.google.com/drive/1fwTC-GRW3X6QiJq6_abJ47On2f3s9e5e) which hopefully makes it easier to try out the library.

@wconnell If you have any notebooks you'd like to add, they can go in the examples folder. I think it'll be best to have them all on Google Colab since you can get it to install all the necessary packages.
